### PR TITLE
Added Subdomain Parameter to Set-AzureRmIoTCentralApp

### PR DESCRIPTION
**Pull request by cle5eland**
## Description

Added the ability to update application subdomain using the Set-AzureRmIoTCentralApp cmdlet.
* Added parameter to update subdomain.
* Added logic to check subdomain availability before updating
* Updated Scenario Tests to test updating subdomain

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
-﻿---
-external help file: Microsoft.Azure.PowerShell.Cmdlets.IotCentral.dll-Help.xml
-Module Name: Az.IotCentral
+---
+external help file: Microsoft.Azure.Commands.IotCentral.dll-Help.xml
+Module Name: AzureRM.IotCentral
```
Comment (cormacpayne) [inline src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md]: @cle5eland it looks like `Az` was changed to `AzureRM` in this file (and the `dll.Help.xml` file above changed). Please make sure that you don't have `AzureRM.IotCentral` installed on your machine when you generate the markdown files -- only the local `Az.IotCentral` module that you build should be imported. Let me know if you have any questions with this.

Comment (cle5eland) [reply src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md]: Makes sense, let me fix.

Comment (cle5eland) [reply src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md]: @cormacpayne should be fixed now

File: src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md [hunk 2]
```diff
@@ -1,93 +1,56 @@
...
-Updates the metadata for an IoT Central Application.
+{{Fill in the Synopsis}}
 
 ## SYNTAX
 
 ### ResourceIdParameterSet (Default)
 ```
-Set-AzureRmIotCentralApp [-DisplayName <String>] [-Tag <Hashtable>] -ResourceId <String> [-AsJob]
- [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+Set-AzIotCentralApp [-DisplayName <String>] [-Subdomain <String>] [-Tag <Hashtable>] -ResourceId <String>
```
Comment (r-delgadillo) [inline src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md]: I believe we should add `sku` and `companyName` to perform the full functionality of our rp-service update api.

Comment (cle5eland) [reply src/IotCentral/IotCentral/help/Set-AzIotCentralApp.md]: @r-delgadillo company name? Also I feel like we intentionally left out Sku. There isn't really any functionality right now--we don't support F1, and you can't update a subless app through the SDK (it's not an azure resource).

File: src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs [hunk 3]
```diff
@@ -53,7 +59,9 @@ public override void ExecuteCmdlet()
         private AppPatch CreateApplicationPatch()
         {
             App existingIotCentralApplication = this.GetApplication();
+            this.EnsureSubdomainAvailabilityOrThrow();
```
Comment (r-delgadillo) [inline src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs]: I don't believe this is needed. rp-service does validation for this

File: src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs [hunk 4]
```diff
@@ -53,7 +59,9 @@ public override void ExecuteCmdlet()
         private AppPatch CreateApplicationPatch()
         {
             App existingIotCentralApplication = this.GetApplication();
```
Comment (r-delgadillo) [inline src/IotCentral/IotCentral/IotCentral/SetAzureRmIotCentralApp.cs]: We shouldn't be getting the application and setting the properties for the patch. Just send the properties that the user wants to update and let ARM do the resource validation & rp-service do the properties validation.

File: src/IotCentral/IotCentral/Common/IotCentralUtils.cs [hunk 5]
```diff
@@ -42,5 +43,13 @@ public static AppPatch CreateAppPatch(App iotCentralApp)
             };
             return copiedIotCenralApp;
         }
+
+        public static void EnsureAvailabilityOrThrow(AppAvailabilityInfo availabilityInfo)
+        {
+            if (availabilityInfo.NameAvailable != true)
```
Comment (cormacpayne) [inline src/IotCentral/IotCentral/Common/IotCentralUtils.cs]: @cle5eland nit:

```cs
if (!availability.NameAvailable)
```

**Comment by maddieclayton** [comment]
@cle5eland Looks like this PR is no longer building after the last commit - can you take a look?

**Comment by cle5eland** [comment]
@maddieclayton yup @cormacpayne  and i are discussing offline, will be resolved as soon as one of my teammates gets out of a meeting

**Comment by cle5eland** [comment]
@cormacpayne @maddieclayton just spoke with @r-delgadillo and he says my current implementation handles the null case correctly. If you guys are good with it then this PR should be good to go.

**Pull request merged**